### PR TITLE
add editing question via api

### PR DESCRIPTION
**Pull request by miltontony**

### CODE REVIEW

**Comment by miltontony** [comment]
@Rizziepit review please

File: unicore/ask/service/schema.py [hunk 1]
```diff
@@ -30,3 +30,36 @@ class QuestionSchema(colander.MappingSchema):
...
     options = Options(validator=validators.options_validator, default=[])
+
+
+class QuestionOptionSchemaPut(colander.MappingSchema):
+    uuid = colander.SchemaNode(
+        colander.String(),
+        validator=validators.option_uuid_validator, missing=None)
+    title = colander.SchemaNode(
+        colander.String(),
+        validator=validators.option_title_validator, missing=None)
```
Comment (Rizziepit) [inline unicore/ask/service/schema.py]: Is a title for a multiple choice option allowed to be None?

File: unicore/ask/service/schema.py [hunk 2]
```diff
@@ -30,3 +30,36 @@ class QuestionSchema(colander.MappingSchema):
...
+
+
+class OptionsPut(colander.SequenceSchema):
+    option = QuestionOptionSchemaPut()
+
+
+class QuestionSchemaPut(colander.MappingSchema):
+    title = colander.SchemaNode(
+        colander.String(),
+        validator=validators.question_title_validator, missing=None)
```
Comment (Rizziepit) [inline unicore/ask/service/schema.py]: `missing=None` here but `nullable=False` on the model

File: unicore/ask/service/schema.py (hunk 1 above)
Comment (miltontony) [inline unicore/ask/service/schema.py]: No, but `put` should allow for partial updates. So if you just need to change a question to a multi-select, you can just submit a put with `{'multiple': True}`

File: unicore/ask/service/schema.py (hunk 2 above)
Comment (Rizziepit) [inline unicore/ask/service/schema.py]: I see that you only set attributes if they are not `None`, but for a `PUT` request all fields are required

File: unicore/ask/service/schema.py (hunk 1 above)
Comment (Rizziepit) [inline unicore/ask/service/schema.py]: Afaik `patch` is the partial update one. I'm fine with using `put` but the docs will have to be clear. Another issue then is that the endpoint does a partial update on the question, but a full update on the options

**Comment by miltontony** [comment]
@Rizziepit hmm..perhaps I should implement patch for partial updates.

**Comment by miltontony** [comment]
@Rizziepit k, I'll ditch the partial updates and implement that on a patch.

**Comment by Rizziepit** [comment]
@miltontony yeah, I think that's better. I've usually left patch out altogether. It tends to get complicated.

**Comment by miltontony** [comment]
@Rizziepit k, I'll just ditch patch altogether then :)

**Comment by miltontony** [comment]
@Rizziepit I've removed the patch stuff and updated the tests.

**Comment by Rizziepit** [comment]
+1

**Pull request merged**